### PR TITLE
Feature: Added force-fetch functionality (#858)

### PR DESCRIPTION
**Pull request by MatteoVoges**
Fixes issue: #858 
## Proposed Changes
* Added a option to have a `fetch_always`-property in your external dependencies.
    If `fetch_always` is set to `true`, this dependency will always get fetched and overwrite existing files (force-fetch).

* Renamed `--force`-flag to `--force-fetch`, so you have to run
    * ` $ kapitan compile --fetch` to fetch (existing files won't get overwritten) or
    * ` $ kapitan compile --force-fetch` to fetch the dependencies and overwrite existing files.

If the property is false or not given, the default-behavior of the fetch-functionality doesn't change.
Also the behavior of `--fetch` and (now:)`--force-fetch` doesn't change either.

**NOTE:** Using `--force` will still work with its default behavior, but is now a _deprecated_ flag and will be removed soon.

## Documentation and testing
Updated flag-name in the docs and in the tests.

### CODE REVIEW

**Comment by MatteoVoges** [comment]
Hi @ramaro @ademariag ,

I applied the changes we discussed, so the PR is ready to review.

But there is one more thing I noticed:
I don't like the way the code for the fetch-functionality is structured. There are multiple if-statements checking the same, and its hard to follow and unterstand the code in general. I would suggest something like this:
```python
[...]
# getting the flag values
if fetch:
    # fetch inventories
    # fetch dependencies
else:
    # fetch targets with fetch_always tag
[...]
```
So with that we would have one if-statement, which indicates if we are fetching or not.

Should I dump my suggestion into this PR or wait until this PR is merged and then create a new one?

**Comment by ramaro** [comment]
@MatteoVoges thanks for getting this done! 

>Should I dump my suggestion into this PR or wait until this PR is merged and then create a new one?
I think doing this in a new PR makes sense to keep things simpler!

**Comment by ramaro** [comment]
This lgtm @MatteoVoges ! Would you mind updating the doc changes and title to force_fetch instead of fetch_always? We can merge straight after that,

**Comment by ramaro** [comment]
@MatteoVoges there's some `fetch_always` usage in docs and code. Particularly because during our discussion, we've agreed to rename to `force_fetch` and this is not reflected for example in `valid_target_obj()`. Unless I'm missing something here 🤔

**Comment by MatteoVoges** [comment]
@ramaro , you're right. I updated it now. Please let me know if there is still something to do here.

File: docs/kap_proposals/kap_1_external_dependencies.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ Git type may also include `ref` and `subdir` parameters as illustrated below:
   source: <git_url>
   subdir: relative/path/in/repository
   ref: <commit_hash/branch/tag>
+  fetch_always: <bool>
```
Comment (ramaro) [inline docs/kap_proposals/kap_1_external_dependencies.md:30]: @MatteoVoges last update here needed

Comment (MatteoVoges) [reply docs/kap_proposals/kap_1_external_dependencies.md]: @ramaro ahh I missed this one. Done with 5b632c26316309f51764a369284d3edb4415867e

Comment (ramaro) [reply docs/kap_proposals/kap_1_external_dependencies.md]: now that was quick! Thanks 👍🏼
